source/common/init/manager_impl.h

```diff
@@ -56,6 +61,12 @@ class ManagerImpl : public Manager, Logger::Loggable<Logger::Id::init> {
 
   // All registered targets
   std::list<TargetHandlePtr> target_handles_;
+  
+  // Corresponding name of registered targets
+  std::vector<std::string> target_names_;
+
+  // Count of target_name
+  std::unordered_map<std::string, uint32_t> target_names_count_;
```

Comment:
I don't understand the purpose of this. Why would somebody want to have multiple init targets registered with the same name?

Reply:
I think no one would intentionally do it :)
But I would also assume duplicated names is allowed somehow.

Reply:
It's probably a bug if somebody does this. I think we should assert that init target names are unique within an init manager, and change this from a map to a set.

Also, actually, it shouldn't be a `std::unordered_`anything, it should be an `absl::flat_hash_set` :)